server/service/integration_mdm_test.go

```diff
@@ -929,6 +929,14 @@ func (s *integrationMDMTestSuite) TestGetMDMCSR() {
 	t := s.T()
 	ctx := context.Background()
 
+	// Validate errors if no private key is set
+	testSetEmptyPrivateKey = true
+	s.uploadAPNSCert([]byte("-----BEGIN CERTIFICATE-----\nZm9vCg==\n-----END CERTIFICATE-----"), http.StatusInternalServerError, "Couldn't upload APNs certificate. Missing required private key. Learn how to configure the private key here: https://fleetdm.com/learn-more-about/fleet-server-private-key")
+
+	r := s.Do("GET", "/api/latest/fleet/mdm/apple/request_csr", getMDMAppleCSRRequest{}, http.StatusInternalServerError)
+	require.Contains(t, extractServerErrorText(r.Body), "Couldn't download signed CSR. Missing required private key. Learn how to configure the private key here: https://fleetdm.com/learn-more-about/fleet-server-private-key")
+	testSetEmptyPrivateKey = false
```

Comment:
You probably want that to be in a `t.Cleanup` call right after setting it to `true`, so that it is properly reset if any of the code in-between fails (so it doesn't affect other tests). Note that a package-level var tends to bite us in the long term, e.g. when multiple tests start using it and setting it in parallel. Might be fine for now, but something to keep  in mind if there's a safer way we can inject the same behavior.